src/adapter/src/coord/sequencer.rs

```diff
@@ -417,86 +417,91 @@ impl<S: Append + 'static> Coordinator<S> {
-                    ))
-                } else {
-                    None
-                };
-
-                self.controller
-                    .storage
-                    .create_collections(vec![(
-                        source_id,
-                        CollectionDescription {
-                            desc: source.desc.clone(),
-                            ingestion: Some(ingestion),
-                            since: None,
-                            status_collection_id,
-                            host_config: Some(source.host_config),
-                        },
-                    )])
-                    .await
-                    .unwrap();
+                    // This is disabled for the moment because it has unusual upper
+                    // advancement behavior.
+                    // See: https://materializeinc.slack.com/archives/C01CFKM1QRF/p1660726837927649
+                    let status_collection_id = if false {
+                        Some(self.catalog.resolve_builtin_storage_collection(
+                            &crate::catalog::builtin::MZ_SOURCE_STATUS_HISTORY,
+                        ))
+                    } else {
+                        None
+                    };
 
-                self.initialize_storage_read_policies(
-                    vec![source_id],
-                    DEFAULT_LOGICAL_COMPACTION_WINDOW_MS,
-                )
-                .await;
+                    self.controller
+                        .storage
+                        .create_collections(vec![(
+                            source_id,
+                            CollectionDescription {
+                                desc: source.desc.clone(),
+                                ingestion: Some(ingestion),
+                                since: None,
+                                status_collection_id,
+                                host_config: Some(source.host_config),
+                            },
+                        )])
+                        .await
+                        .unwrap();
+
+                    self.initialize_storage_read_policies(
+                        vec![source_id],
+                        DEFAULT_LOGICAL_COMPACTION_WINDOW_MS,
+                    )
+                    .await;
+                }
                 Ok(ExecuteResponse::CreatedSource { existed: false })
             }
-            Err(AdapterError::Catalog(catalog::Error {
-                kind: catalog::ErrorKind::ItemAlreadyExists(_),
-                ..
-            })) if plan.if_not_exists => Ok(ExecuteResponse::CreatedSource { existed: true }),
+            //TODO(petrosagg): figure this out
+            // Err(AdapterError::Catalog(catalog::Error {
+            //     kind: catalog::ErrorKind::ItemAlreadyExists(_),
+            //     ..
+            // })) if plan.if_not_exists => Ok(ExecuteResponse::CreatedSource { existed: true }),
```

Comment:
Since the user is only typing `CREATE SOURCE` once, the `if_not_exists` flag should probably propagate to all the sub-sources, so every plan here should have the same value for `if_not_exists`. However, it's not clear to me what we should do if some subset of the sources exists and some other subset don't exist. 

For example, what happens if we have a source `s1`, with subsources `ss1` and `ss2`. Then a user tries to create a new source `s1` with subsources `ss2` and `ss3` with the flag `if_not_exists` set to `true`?

I personally think that `CREATE IF NOT EXISTS` (CINE) is a bad pattern in general, this being another reason, and we should just remove it from `CREATE SOURCE`. Any thoughts @benesch on removing CINE from sources?

Reply:
I think users use CINE as a care-free way to re-run DDL statements and ensure that a certain end state is reached

Reply:
The problem with CINE is that you don't actually know anything about the end state, just the name of the source. For example the source that exists may be ingesting data from a completely different source than what you just tried to create with and you have know way of knowing except by checking after running the `CREATE SOURCE` command.

It would make sense to me for `CINE` to return OK if there's an existing source with the same name AND it has all the same subsources that we are trying to create. If the main source exists, but none of the subsources exists, then I feel like we'd definitely be in a state the user wasn't expecting and we should respond OK.

Reply:
Unless I'm misunderstanding, I don't think the proposal of having `if_not_exists` only apply to the main source would allow users to re-run the same DDL statement. The second run would have colliding subsources and error.

Reply:
I was imagining having the logic do something like "if `if_not_exists` is set, and the main source exists, do nothing regardless of subsources"